core/src/ast/expr_decoder.rs

```diff
@@ -0,0 +1,323 @@
+            low,
+            high,
+        } => match negated {
+            true => format!(
+                "{:} NOT BETWEEN {:} AND {:}",
+                decode(&*expr),
+                decode(&*low),
+                decode(&*high)
+            ),
+
+            false => format!(
+                "{:} BETWEEN {:} AND {:}",
+                decode(&*expr),
+                decode(&*low),
+                decode(&*high)
+            ),
+        },
+        Expr::UnaryOp { op, expr } => format!("{:}{:}", op, decode(&*expr)),
+        Expr::Cast { expr, data_type } => {
+            format!("CAST({:} AS {:})", decode(&*expr), data_type)
+        }
+        Expr::Extract { field, expr } => {
+            format!("EXTRACT({:} FROM \"{:}\")", field, decode(&*expr))
+        }
+        Expr::Nested(expr) => format!("todo:Nested({:})", decode(&*expr)),
+        Expr::Literal(s) => match s {
+            AstLiteral::Boolean(b) => format!("{:}", b),
+            AstLiteral::Number(d) => format!("{:}", d),
+            AstLiteral::QuotedString(qs) => format!("\"{:}\"", qs),
+            AstLiteral::Null => "Null".to_string(),
+            AstLiteral::Interval { .. } => "Interval not implemented yet..".to_string(),
+        },
+        Expr::TypedString { data_type, value } => format!("{:}(\"{:}\")", data_type, value),
+        Expr::Case {
+            operand,
+            when_then,
+            else_result,
+        } => {
+            let mut str = match operand {
+                Some(s) => format!("CASE {:}", decode(s)),
+                None => "CASE ".to_string(),
+            };
+            for (_when, _then) in when_then {
+                str += format!("\nWHEN {:} THEN {:}", decode(_when), decode(_then)).as_str();
+            }
+
+            match else_result {
+                Some(s) => str += format!("\nELSE {:}", decode(s)).as_str(),
+                None => str += "", // no operation?
+            };
+            str + "\nEND"
+        }
+        Expr::Aggregate(a) => match &**a {
+            Aggregate::Count(c) => match c {
+                CountArgExpr::Expr(e) => format!("Count({:})", decode(e)),
+                CountArgExpr::Wildcard => "Count(*)".to_string(),
+            },
+            Aggregate::Sum(e) => format!("Sum({:})", decode(e)),
+            Aggregate::Max(e) => format!("Max({:})", decode(e)),
+            Aggregate::Min(e) => format!("Min({:})", decode(e)),
+            Aggregate::Avg(e) => format!("Avg({:})", decode(e)),
+        },
+        Expr::Function(f) => {
+            format!("{:}(todo:args)", f)
+        }
+        // todo's...  these require enum query..
+        //Expr::InSubquery {expr, subquery, negated} => format!("InSubquery({:}, subquery:{:}, negated: {:})", decode(*expr),  *subquery, negated),
+        //Expr::Exists(q) => format!("Exists({:})", *query),
+        //Expr::Subquery(q) => format!("Subquery({:})", *query),
+        _ => format!("Unimplemented Decode Expression: {:#?}", expr),
```

Comment:
This looks good start!
We don't need to handle all the display impl in here.
As you did for `Expr::Function`,
```rust
Expr::Function(f) => {
    format!("{:}(todo:args)", f)
}
```

How about wrapping up something like this for now?
```rust
Expr::Exists(q) => "EXISTS(..)",
```

Then, we can keep this SQL display task in following PRs.

Reply:
sounds good to me.